### PR TITLE
One file per test Recording infrastructure

### PR DESCRIPTION
**Pull request by amarzavery**
There are too many files to be reviewed:
Important files to look at:
*  [test/framework/cli-test.js](https://github.com/Azure/azure-xplat-cli/pull/1642/files#diff-b3abb1286bae61323d158be618d89bcb)
*  [test/framework/preamble.template](https://github.com/Azure/azure-xplat-cli/pull/1642/files#diff-f73b73089304ad3ab106c83046985340)
*  [test/framework/test-logger.js](https://github.com/Azure/azure-xplat-cli/pull/1642/files#diff-602f93e8f617bbc2dadfa05075f9caef)
*  [test/framework/xcli-test-reporter.js](https://github.com/Azure/azure-xplat-cli/pull/1642/files#diff-234887203d94c19bdf44eb62aaaac095)

Most of the other files would be inside the test/recordings/**{suite-name}/{full-test-title.nock.js}**

### CODE REVIEW

File: lib/commands/asm/network._js [hunk 1]
```diff
@@ -740,12 +740,12 @@ exports.init = function(cli) {
...
-      if (utils.ignoreCaseEquals(site.Name, vnetInput.name)) {
-        throw new Error(util.format($('A virtual network with name %s already exists'), vnetInput.name));
-      }
+    if (vnetConfiguration.VirtualNetworkSites) {
+      vnetConfiguration.VirtualNetworkSites.forEach(function (site) {
+        if (utils.ignoreCaseEquals(site.Name, vnetInput.name)) {
+          throw new Error(util.format($('A virtual network with name %s already exists'), vnetInput.name));
+        }
+      });
     }
```
Comment (amarzavery) [inline lib/commands/asm/network._js]: this fixes issue #1632

**Comment by yugangw-msft** [comment]
lgtm, would be great to update wiki of how to author new test, and the recommended naming of the test, and the mapping with the the record file

**Comment by amarzavery** [comment]
Fixes #1635, #1632, #1514

File: lib/commands/asm/vm/vmclient.js [hunk 2]
```diff
@@ -587,11 +587,20 @@ _.extend(VMClient.prototype, {
                 // which will be already linked with a vhd
                 for (var i = 0; i < role[key].length; i++) {
                   delete role[key][i].mediaLink;
-                  // delete role[key][i].sourceMediaLink; property is deprecated
+                  //IOType is impicitly determined from the media link hence it is not required
+                  //when a vm is created from the exported json file
+                  if (role[key][i].iOType) {
+                    delete role[key][i].iOType;
+                  }
```
Comment (amarzavery) [inline lib/commands/asm/vm/vmclient.js]: Fixes #1635, #1514

**Pull request merged**